tests/test_benchmark.py

```diff
@@ -236,32 +235,51 @@ def __init__(self, name, engine, args=[OPTIMIZATIONS], binaryen_opts=[]):
 
   def build(self, parent, filename, args, shared_args, emcc_args, native_args, native_exec, lib_builder, has_output_parser):
     suffix = filename.split('.')[-1]
-    cheerp_temp = filename + '.cheerp.' + suffix
+    cheerp_temp = filename[:-len(suffix) - 1] + '.cheerp.cpp'
     code = open(filename).read()
-    if 'int main()' in code:
-      main_args = ''
-    else:
-      main_args = 'argc, (%(const)s char**)argv' % {
-        'const': 'const' if 'const char *argv' in code else ''
-      }
     open(cheerp_temp, 'w').write('''
       %(code)s
-      void webMain() {
-        // TODO: how to read from commandline?
-        volatile int argc = 2;
-        typedef char** charStarStar;
-        volatile charStarStar argv;
-        argv[0] = "./cheerp.exe";
-        argv[1] = "%(arg)s";
-        volatile int exit_code = main(%(main_args)s);
-      }
-    ''' % {
-      'arg': args[-1],
+#include <cheerp/client.h>
+namespace client
+{
+    extern client::TArray<client::String>& args;
+}
+class [[cheerp::genericjs]] Arguments
+{
+public:
+    static int getArgumentsCount()
+    {
+        return client::args.get_length();
+    }
+    static int getArgumentLength(int arg)
+    {
+        return client::args[arg]->get_length();
+    }
+    static void copyArgument(int arg, char* buf)
+    {
+        client::String* a = client::args[arg];
+        for(int i=0;i<a->get_length();i++)
+            *(buf++) = a->charCodeAt(i);
+        *buf = 0;
+    }
+};
+void webMain() {
+    int arg1Len = Arguments::getArgumentLength(0);
```

Comment:
This is fine to land, but fyi I am getting rid of the dependence on commandline arguments very soon - it doesn't work or is hard to use in some cases (e.g. adding commandline arguments for the web), and sometimes I send people a build and they forget to run it with an argument. Instead I'll just hardcode argc/argv into the benchmark main file before compiling it.

Reply:
No problem about that, when you migrate to hardcoding the parameters just remove this code.